frontend/src/metabase-types/api/pagination.ts

```diff
@@ -0,0 +1,15 @@
+export interface PaginationInput {
+  page?: number;
+  pageSize?: number;
+}
+
+export interface PaginationRequest {
+  limit?: number;
+  offset?: number;
+}
+
+export interface PaginationResponse {
+  limit: number;
+  offset: number;
```

Comment:
this can also be null sometimes, right?
```suggestion
  limit: number | null;
  offset: number | null;
```

Reply:
I still don't know exactly when, tbh.
If I set this to nullable type, it breaks a type check.

Is it too bad if we update types later as we go, i.e. once we reach the need for it to be nullable?

Reply:
Fixing it is not an issue.
I noticed some weird behavior which tells me that we don't really understand how some of these endpoints work, and I wanted to understand that first, and only then adjust the types.

For example, I used Insomnia to call `GET /api/task` without any params, but still got this response:
```
{
  total: number,
  limit: null,
  offset: null,
  data: object[]
}
```

Before hitting this, I was convinced that the backend omits `total`, `limit` and `offset` unless you explicitly send either `limit` of `offset` in the request, but there might be some endpoints that are wired differently.

Reply:
Types have been properly updated.

```ts
export interface PaginationRequest {
  limit?: number | null;
  offset?: number | null;
}

export interface PaginationResponse {
  limit: number | null;
  offset: number | null;
  total: number;
}
```